x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportPutDataFrameTransformAction.java

```diff
@@ -223,17 +221,16 @@ private void handlePrivsResponse(String username,
         if (privilegesResponse.isCompleteMatch()) {
             putDataFrame(config, listener);
         } else {
-            XContentBuilder builder = JsonXContent.contentBuilder();
-            builder.startObject();
-            for (ResourcePrivileges index : privilegesResponse.getIndexPrivileges()) {
-                builder.field(index.getResource());
-                builder.map(index.getPrivileges());
-            }
-            builder.endObject();
+            List<String> indices = privilegesResponse.getIndexPrivileges()
+                .stream()
+                .map(ResourcePrivileges::getResource)
+                .collect(Collectors.toList());
 
-            listener.onFailure(Exceptions.authorizationError("Cannot create data frame transform [{}]" +
-                    " because user {} lacks permissions on the indices: {}",
-                config.getId(), username, Strings.toString(builder)));
+            listener.onFailure(Exceptions.authorizationError(
+                "Cannot create data frame transform [{}] because user {} lacks the all the required permissions for indices: [{}]",
```

Comment:
1. `... lacks the all the required...`: remove the first `the`
2. nit: No need to use `Strings.collectionToCommaDelimitedString(indices)`. You could just use `indices` as long as you remove the square brackets around the placeholder. A list's `toString` method will comma-separate and wrap in square brackets on its own.